test/new-e2e/cws/e2e_cws_test.go

```diff
@@ -0,0 +1,150 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package cws
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	cws "github.com/DataDog/datadog-agent/test/new-e2e/cws/lib"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner/parameters"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/params"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2params"
+)
+
+type agentSuite struct {
+	e2e.Suite[e2e.AgentEnv]
+	apiClient     cws.MyAPIClient
+	signalRuleID  string
+	agentRuleID   string
+	dirname       string
+	filename      string
+	testID        string
+	desc          string
+	agentRuleName string
+	policies      string
+}
+
+func TestAgentSuite(t *testing.T) {
+	agentConfig := "hostname: momar-e2e-test-host\nlog_level: DEBUG\nlogs_enabled: true"
+	e2e.Run(t, &agentSuite{}, e2e.AgentStackDef(
+		[]ec2params.Option{
+			ec2params.WithName("cws-e2e-tests"),
+		},
+		agentparams.WithAgentConfig(agentConfig),
+		agentparams.WithSecurityAgentConfig("runtime_security_config:\n  enabled: true"),
+		agentparams.WithSystemProbeConfig("system_probe_config:\n  enabled: true\n  log_level: trace\nruntime_security_config:\n  enabled: true\nlog_patterns:\n  - module.APIServer.*"),
+		agentparams.WithVersion("7.46.0"),
+	), params.WithDevMode())
```

Comment:
<!-- Warning highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes and they are blockers. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question. -->
**🔨 warning**
We discourage using `WithDevMode` on the CI. You can keep it by enabling it based on an env var for example, to keep using it by default locally